client/db/bolt/db.go

```diff
@@ -818,18 +819,44 @@ func newestBuckets(master *bbolt.Bucket, n int, timeKey []byte, filter func(*bbo
 	return idx.pairs
 }
 
-// makeTopLevelBuckets creates a top-level bucket for each of the provided keys,
-// if the bucket doesn't already exist.
-func (db *boltDB) makeTopLevelBuckets(buckets [][]byte) error {
-	return db.Update(func(tx *bbolt.Tx) error {
+// init sets the database version and creates top-level bucket
+// for each of the provided keys, if the bucket doesn't already exist.
+func (db *boltDB) init(buckets [][]byte) error {
+	err := db.Update(func(tx *bbolt.Tx) error {
 		for _, bucket := range buckets {
 			_, err := tx.CreateBucketIfNotExists(bucket)
 			if err != nil {
 				return err
 			}
 		}
+
+		// Set the db version if it does not already exist.
+		bucket := tx.Bucket(appBucket)
+		if bucket == nil {
+			return fmt.Errorf("app bucket not found")
+		}
+		versionB := bucket.Get(versionKey)
+		if versionB == nil {
+			versionB = encode.Uint32Bytes(uint32(initialVersion))
```

Comment:
A log here to the effect that a newdb at version X is being created would be nice.